stock_product_pack/models/stock_rule.py

```diff
@@ -0,0 +1,28 @@
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, models
+
+
+class ProcurementGroup(models.Model):
+    _inherit = "procurement.group"
+
+    @api.model
+    def run(self, procurements):
+        """ If 'run' is called on a pack product storable.
+         we remove the procurement with this product pack.
+        """
+        for procurement in procurements:
+            if (
+                procurement.product_id
+                and procurement.product_id.pack_ok
+                and procurement.product_id.dont_create_move
+                and procurement.product_id.pack_type == "detailed"
+            ):
+                procurements.remove(procurement)
+                sale_order_id = procurement.values.get("sale_line_id")
+                if sale_order_id:
+                    self.env["sale.order.line"].browse(
+                        sale_order_id
+                    ).qty_delivered = procurement.product_qty
```

Comment:
This is triggering the invoiceability of the line before it should, so it's not a valid option.

Reply:
you're right Pedro, it would affect if invoice policy is "delivered".
Maybe we can try to make the same approach as boms here, regarding the delivery of the components this quantity would be updated. We try to avoid so because it was more complex but I didn't think about the invoiceability.

What do you think? Any other suggestion?

Reply:
I still think that changing the visibility of the smart-buttons should be enough, isn't it?

Reply:
I don't think so:
a) if you use consumable: stock will be visible, the pack would still travel to the picking and you will move a product that is not a product. Also, on ecommerce it wont consider stock (so actually you would sell packs of products without stock). It's heavier to fix that on ecomerce:
https://github.com/odoo/odoo/blob/13.0/addons/website_sale_stock/static/src/xml/website_sale_stock_product_availability.xml#L6
https://github.com/odoo/odoo/blob/9b91f9d1bb41d05f445d745a9b8e5fa63edaf2a9/addons/website_sale_stock/models/sale_order.py#L18
https://github.com/odoo/odoo/blob/9b91f9d1bb41d05f445d745a9b8e5fa63edaf2a9/addons/website_sale_stock/static/src/js/variant_mixin.js#L49

b) if you use it a service: stock could be visible (we also need to patch the method as proposed [here](https://github.com/OCA/product-pack/pull/41), same issues on ecommerce and the product wont be set as delivered automatically. You would need to put manually the delivered quantity

Reply:
Then what is your suggestion?

Reply:
My suggestion is go for this pr with two options:
1. keep as is but adding a warning or known issue that if the user decides too use the new option "dont_create_move" with products with invoice policy "to deliver", then the "invoiceability" won't be accurate.
2. improve this PR and try to set the delivery quantity from the delivered pack lines, similar to what is done from product phantom boms

For me we can start with "1" and keep improvement on the road-map, but if you prefer "2" we can go for that.

What do you think?

Reply:
For me 1 is not acceptable. Let's go for 2.